lib/gplt.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+require "thor"
+require "./tests/helpers/perf_run_helper"
+
+# Add pry support when available
+begin
+  require "pry"
+rescue LoadError # rubocop: disable Lint/HandleExceptions
+  # do nothing
+end
+
+module GPLT
+  class Error < StandardError; end
+
+  # Class to define CLI subcommands
+  class CLI < Thor
+    extend PerfRunHelper
+
+    default_test_type = "apples to apples"
+
+    option :test_type, default: default_test_type
+    option :baseline, required: true
+    option :results_dir, required: true
+
+    desc "validate2baseline",
+         "validate perf results against baseline data"
+
+    long_desc <<-LONGDESC
+      `validate2baseline` validates performance results data against baseline data.
+
+      You must specify a --baseline for the baseline data and a --results_dir
+      containing the performance results to compare for validation.
+
+      The --test_type is optional and defaults to "#{default_test_type}"
+
+      A passing state will echo "PASS" to STDOUT and exit with 0.
+      A failing state will echo "FAIL" to STDOUT and exit with 1.
+    LONGDESC
+
+    def validate2baseline
+      if CLI.validate_results_to_baseline(options[:results_dir], options[:baseline], options[:test_type])
+        puts "PASS"
```

Comment:
unless I am missing something... it doesn't actually trigger the code to do the comparison...

Reply:
I think you are missing something. :-)
You should be able to test this for yourself on the command line:
```
$ bundle exec  bin/gplt validate2baseline --baseline=2019.1.0 --results-dir=$MYRESULTS
Getting mean response time from /home/john/Downloads/results/perf/PERF_1571758219/ip-10-227-3-92.amz-dev
.puppet.net/root/gatling-puppet-load-test/simulation-runner/results/PerfTestLarge-1571758226846/js/globa
l_stats.json
Baseline result returned from BigQuery: [{:avg_cpu=>20, :avg_mem=>2630938, :avg_disk_write=>22063, :avg_
response_time=>644, :process_bolt_server_avg_cpu=>nil, :process_bolt_server_avg_mem=>nil, :process_ace_s
erver_avg_cpu=>nil, :process_ace_server_avg_mem=>nil, :process_puppetdb_avg_cpu=>3, :process_puppetdb_av
g_mem=>734776, :process_console_services_release_avg_cpu=>2, :process_console_services_release_avg_mem=>
549411, :process_orchestration_services_release_avg_cpu=>6, :process_orchestration_services_release_avg_
mem=>595188, :process_puppet_server_release_avg_cpu=>6, :process_puppet_server_release_avg_mem=>3706311}
]
...
Result 'avg_cpu' is outside tolerances: baseline: 20; result: 11; variance: -45.0%
Result 'avg_response_time' is outside tolerances: baseline: 644; result: 391; variance: -39.29%
Result 'process_puppetdb_avg_cpu' is outside tolerances: baseline: 3; result: 0; variance: -100.0%
Result 'process_console_services_release_avg_cpu' is outside tolerances: baseline: 2; result: 3; variance: 50.0%
Result 'process_orchestration_services_release_avg_cpu' is outside tolerances: baseline: 6; result: 4; variance: -33.33%
Result 'process_puppet_server_release_avg_cpu' is outside tolerances: baseline: 6; result: 0; variance:
-100.0%
FAIL
$ echo $?
1
```

Reply:
looks like thor does it under the hood.

Reply:
ok i see it.  You extend perf_runner and thus it brings validate_results_to_baseline into the class...
Any reason why you didn't require it instead and then call into it?